### PR TITLE
✨ Use @percy/cli

### PR DESCRIPTION
**Pull request by wwilsman**
## What is this?

This PR brings this SDK up-to-date with the rest of our SDKs by upgrading it to use the new `@percy/cli`.

Previously, this SDK was a small wrapper around running `@percy/agent`'s `snapshot` command pointed at Gatsby's output directory. With the new CLI toolchain, we can make this SDK much more feature rich and robust.

- Percy can easily be disabled like other SDKs
- More advanced snapshot options can be accepted by the new CLI
- Page paths are derived directly from Gatsby rather than rely on output file structure
- Updates to core internals do not need to propagate down through SDK updates
- All of the advantages that come with `@percy/cli` over `@percy/agent`

The whole repo was also updated to match the rest of our ecosystem with GitHub actions & templates and other meta config like eslint and test coverage.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  commit-message:
+    prefix: ⬆
+- package-ecosystem: github-actions
+  directory: /
+  schedule:
+    interval: weekly
+  commit-message:
+    prefix: ⬆👷
+  labels:
+    - ⬆⬇ dependencies
```
Comment (Robdel12) [inline .github/dependabot.yml:18]: These seem encoded weirdly, but might be fine?

File: gatsby-node.js [hunk 2]
```diff
@@ -1,25 +1,93 @@
...
+  // auto-start the activity
+  activity.start();
+  return context;
+}
+
+// After building, start Percy, query pages, serve build output, and take snapshots
+exports.onPostBuild = async ({ reporter, graphql, store }, {
+  // gatsby specific options
+  query = '{ allSitePage { nodes { path } } }',
+  resolvePages = ({ allSitePage }) => {
```
Comment (Robdel12) [inline gatsby-node.js:42]: This is 🔥 over what we had previously. Super excited about that!

**Review by Robdel12 — APPROVED** [review]
🏁 The final SDK to go CLI!

File: gatsby-node.js [hunk 3]
```diff
@@ -1,25 +1,93 @@
...
+    throw new Error('Could not find query data at `allSitePage.nodes`. ' + (
+      'Fix the custom `query` or provide a custom `resolvePages` function.'));
+  },
+  // static snapshot options
+  ...staticOptions
+}) => {
+  let activity = activityHelper('Percy', reporter);
+
+  // lazily required to not bloat the process when not building
+  let { serve, mapStaticSnapshots } = require('@percy/cli-snapshot/dist/utils');
```
Comment (habib-deriv) [inline gatsby-node.js:53]: "gatsby-plugin-percy" threw an error while running the onPostBuild lifecycle:

Package subpath './dist/utils' is not defined

**Review by habib-deriv — COMMENTED** [review]
Not working!

"gatsby-plugin-percy" threw an error while running the onPostBuild lifecycle:

Package subpath './dist/utils' is not defined

**Comment by wwilsman** [comment]
Hey @habib-deriv!

This PR was working when merged, so commenting directly on it is unnecessary and unproductive. In the future, please open a new issue for new bugs rather than commenting on 10 month old, outdated, PRs (which can't be reopened anyway since it's not an issue to begin with). 😉 

Since this package was last updated though, the upstream dependencies have changed significantly (there are beta dependencies that are now deprecated). We're working on that though! You can expect this SDK to be updated and start working again within the next week or so. 👍 

In the meantime, the [snapshot command](https://docs.percy.io/docs/cli-snapshot#static-directory) will work just fine when pointed at your Gatsby build output:

```sh
$ percy snapshot ./public
```

**Pull request merged**